addons/delivery/models/stock_move.py

```diff
@@ -19,8 +19,8 @@ def _cal_move_weight(self):
 
     def _get_new_picking_values(self):
```

Comment:
```python
def _get_new_picking_values(self):
        vals = super(StockMove, self)._get_new_picking_values()
        vals['carrier_id'] = self.rule_id.propagate_carrier and self.group_id.sale_id.carrier_id
        return vals
```

Reply:
Thanks for your suggestion

Reply:
`vals['carrier_id'] = any(self.rule_id.propagate_carrier) and self.group_id.sale_id.carrier_id`